### PR TITLE
Added flowLabel, excludeLabelFromRouter flow fields + relabel output

### PR DESCRIPTION
**Pull request by petergvizd**
| Q               | A
| --------------- | ---
| Bug fix?        | no
| New feature?    | yes
| API breaks?     | no
| Deprecations?   | no
| License         | Apache 2.0

### What's in this PR?
- Added flowLabel, excludeLabelFromRouter flow fields
- Added relabel output

### Why?
- To handle timeout log lines the same as normal logs with concat plugin: https://github.com/fluent-plugins-nursery/fluent-plugin-concat
- To have option for setting up global filters applied after flow filters.
```
apiVersion: logging.banzaicloud.io/v1beta1
kind: ClusterOutput
metadata:
  name: final-relabel
spec:
  relabel:
    label: '@final-flow'
---
apiVersion: logging.banzaicloud.io/v1beta1
kind: Flow
metadata:
  name: serviceFlow1
  namespace: namespace1
spec:
  filters: []
  globalOutputRefs:
  - final-relabel
  match:
  - select:
      labels:
        app: service1
---
apiVersion: logging.banzaicloud.io/v1beta1
kind: Flow
metadata:
  name: serviceFlow2
  namespace: namespace2
spec:
  filters: []
  globalOutputRefs:
  - final-relabel
  match:
  - select:
      labels:
        app: service2
---
apiVersion: logging.banzaicloud.io/v1beta1
kind: ClusterFlow
metadata:
  name: final-flow
spec:
  flowLabel: '@final-flow'
  excludeLabelFromRouter: true
  filters: []
```

### Checklist

- [x] Code meets the [Developer Guide](https://github.com/banzaicloud/developer-guide)
- [x] User guide and development docs updated (if needed)
- [x] Related Helm chart(s) updated (if needed)

### CODE REVIEW

File: pkg/sdk/logging/model/types/flow.go [hunk 1]
```diff
@@ -67,6 +67,9 @@ type Flow struct {
 
 	// Fluentd label
 	FlowLabel string `json:"-"`
+
+	//Flag if add flow to label_router
```
Comment (siliconbrain) [inline pkg/sdk/logging/model/types/flow.go:71]:
```suggestion
	// Flag whether to exclude flow from label_router
```

File: pkg/sdk/logging/model/types/flow.go [hunk 2]
```diff
@@ -104,8 +107,11 @@ func (f *Flow) WithOutputs(output ...Output) *Flow {
...
-func NewFlow(matches []FlowMatch, id, name, namespace string) (*Flow, error) {
-	flowLabel, err := calculateFlowLabel(matches, name, namespace)
+func NewFlow(matches []FlowMatch, id, name, namespace, flowLabel string, excludeLabelFromRouter bool) (*Flow, error) {
+	var err error
+	if flowLabel == "" {
+		flowLabel, err = calculateFlowLabel(matches, name, namespace)
+	}
 	if err != nil {
 		return nil, err
 	}
```
Comment (siliconbrain) [inline pkg/sdk/logging/model/types/flow.go:117]:
```suggestion
	if flowLabel == "" {
		var err error
		flowLabel, err = calculateFlowLabel(matches, name, namespace)
		if err != nil {
			return nil, err
		}
	}
```

Comment (petergvizd) [reply pkg/sdk/logging/model/types/flow.go]: @siliconbrain suggestions added

**Comment by petergvizd** [comment]
Hello guys, how it looks like with this PR?

**Comment by petergvizd** [comment]
Hello guys @siliconbrain @tarokkk @ahma , what can I do for merge of this PR?

**Comment by sebastiangaiser** [comment]
Any updates on this?

**Comment by pepov** [comment]
Hey @petergvizd! Can you imagine a situation where you want to set the flowlabel manually, but don't want to exclude it from the router? I'm a bit hesitant to add a configuration flag, that we always have to set to something that should be the default.
So my suggestion here would be to either remove that extra field, or leave it there, but change it from exclude to include and leave it to default false.
Fantastic job by the way!

**Comment by pepov** [comment]
Also could you please rebase? (and sorry for the extreme delay to respond!)

**Comment by pepov** [comment]
One more thing, you are welcome to discuss this topic on our discord channel:
https://discord.gg/aAZdQVEz

**Comment by pepov** [comment]
Hey @petergvizd or @sebastiangaiser are you still interested in this?

**Comment by gvizdpeter** [comment]
Hello, @pepov I am interested, but unfortunately PR was created from company GH account, so I am no longer getting notifications. I will try to prepare your suggestions.

**Comment by pepov** [comment]
@gvizdpeter does it help if I rebase and resolve conflicts for you?

**Comment by gvizdpeter** [comment]
Hello @pepov , as I do not have access to this GH account anymore I created new PR https://github.com/kube-logging/logging-operator/pull/1294 with resolved conflicts, and going to prepare your suggestions.

**Pull request closed without merging**